substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -0,0 +1,1591 @@
+	///
+	/// Currently the ready transaction is returned if it exists for the most recently notified best
+	/// block (for which maintain process was accomplished).
+	// todo [#5491]: api change: we probably should have at here?
+	fn ready_transaction(&self, tx_hash: &TxHash<Self>) -> Option<Arc<Self::InPoolTransaction>> {
+		let most_recent_view = self.view_store.most_recent_view.read();
+		let result = most_recent_view
+			.map(|block_hash| self.view_store.ready_transaction(block_hash, tx_hash))
+			.flatten();
+		log::trace!(
+			target: LOG_TARGET,
+			"[{tx_hash:?}] ready_transaction: {} {:?}",
+			result.is_some(),
+			most_recent_view
+		);
+		result
+	}
+
+	/// Returns an iterator for ready transactions at a specific block, ordered by priority.
+	fn ready_at(&self, at: <Self::Block as BlockT>::Hash) -> PolledIterator<ChainApi> {
+		let (_, result) = self.ready_at_internal(at);
+		result
+	}
+
+	/// Returns an iterator for ready transactions, ordered by priority.
+	///
+	/// Currently the set of ready transactions is returned if it exists for the most recently
+	/// notified best block (for which maintain process was accomplished).
+	fn ready(&self) -> ReadyIteratorFor<ChainApi> {
+		self.view_store.ready()
+	}
+
+	/// Returns a list of future transactions in the pool.
+	///
+	/// Currently the set of future transactions is returned if it exists for the most recently
+	/// notified best block (for which maintain process was accomplished).
+	fn futures(&self) -> Vec<Self::InPoolTransaction> {
+		self.view_store.futures()
+	}
+
+	/// Returns a set of ready transactions at a given block within the specified timeout.
+	///
+	/// If the timeout expires before the maintain process is accomplished, a best-effort
+	/// set of transactions is returned (refer to `ready_at_light`).
+	fn ready_at_with_timeout(
+		&self,
+		at: <Self::Block as BlockT>::Hash,
+		timeout: std::time::Duration,
+	) -> PolledIterator<ChainApi> {
+		self.ready_at_with_timeout_internal(at, timeout)
+	}
+}
+
+impl<Block, Client> sc_transaction_pool_api::LocalTransactionPool
+	for ForkAwareTxPool<FullChainApi<Client, Block>, Block>
+where
+	Block: BlockT,
+	<Block as BlockT>::Hash: Unpin,
+	Client: sp_api::ProvideRuntimeApi<Block>
+		+ sc_client_api::BlockBackend<Block>
+		+ sc_client_api::blockchain::HeaderBackend<Block>
+		+ sp_runtime::traits::BlockIdTo<Block>
+		+ sp_blockchain::HeaderMetadata<Block, Error = sp_blockchain::Error>,
+	Client: Send + Sync + 'static,
+	Client::Api: sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>,
+{
+	type Block = Block;
+	type Hash = ExtrinsicHash<FullChainApi<Client, Block>>;
+	type Error = <FullChainApi<Client, Block> as graph::ChainApi>::Error;
+
+	fn submit_local(
+		&self,
+		_at: Block::Hash,
+		_xt: sc_transaction_pool_api::LocalTransactionFor<Self>,
+	) -> Result<Self::Hash, Self::Error> {
+		//todo [#5493]
+		//looks like view_store / view needs non async submit_local method ?.
+		unimplemented!();
```

Comment:
I think crash for `unimplemented` function is a little bit easier to find during initial tests (keep in mind this is experimental and intended for preliminary testing by teams) then some error.

It is planned as follow up: https://github.com/paritytech/polkadot-sdk/issues/5493

Reply:
nevermind, if you don't like `unimplemented` here, I've added error message and returned the error:
0b715528bb4

I just hope it will not lead to problems that will consume unnecessary debugging time during testing.